python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -39,7 +39,11 @@
 from .id_map import NoopMap, IdMap, map_node_ids
 from .utils import multiprocessing_data_read, ExtMemArrayMerger, partition_graph
 
-def parse_node_data(in_file, feat_ops, label_ops, node_id_col, read_file):
+FEATURE_STORE_PRECISION_FP32 = "fp32"
+FEATURE_STORE_PRECISION_FP16 = "fp16"
+
+def parse_node_data(in_file, feat_ops, label_ops, node_id_col,
+                    read_file, feat_dtype=np.float32):
```

Comment:
i would prefer that you create a feature operator that converts the data type. in this way, we don't need to pass anything argument to any of the functions.

Reply:
in this way, users can define output data type in the JSON file to define the output data type for each feature.